sdk/src/connections/target/adt7410_sensor.cpp

```diff
@@ -29,33 +29,96 @@
  * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
-#include "temp_sensor.h"
+#include "adt7410_sensor.h"
 
 #include <errno.h>
 #include <fcntl.h>
+#include <glog/logging.h>
+#include <linux/fs.h>
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
 
-int temp_sensor_read(temp_sensor *t, float *temp_val) {
+#define TEMPERATURE_MSB_REG 0x0
+#define TEMPERATURE_LSB_REG 0x1
+#define CONFIGURATION_REG 0x3
+
+using namespace aditof;
+
+struct ADT7410::ImplData {
+    temp_sensor tDev;
+    std::string driverPath;
+    int i2c_address;
+};
+
+ADT7410::ADT7410(const std::string &name, const std::string &driver_path,
+                 int i2c_address)
+    : m_implData(new ImplData), m_name(name) {
```

Comment:
Why not use initializer list for all members?

Reply:
Which member isn't initialized?

Reply:
this is actually ok, my bad :)